src/Explorer/Panes/AddDatabasePanel/AddDatabasePanel.tsx

```diff
@@ -0,0 +1,342 @@
+    }
+
+    createDatabase(createDatabaseParams).then(
+      () => {
+        _onCreateDatabaseSuccess(offerThroughput, startKey);
+      },
+      (error: string) => {
+        _onCreateDatabaseFailure(error, offerThroughput, startKey);
+      }
+    );
+  };
+
+  const _onCreateDatabaseSuccess = (offerThroughput: number, startKey: number): void => {
+    setIsExecuting(false);
+    closePanel();
+    container.refreshAllDatabases();
+    const addDatabasePaneSuccessMessage = {
+      ...addDatabasePaneMessage,
+      offerThroughput,
+    };
+    TelemetryProcessor.traceSuccess(Action.CreateDatabase, addDatabasePaneSuccessMessage, startKey);
+  };
+
+  const _onCreateDatabaseFailure = (error: string, offerThroughput: number, startKey: number): void => {
+    setIsExecuting(false);
+    const errorMessage = getErrorMessage(error);
+    setFormErrors(errorMessage);
+    setFormErrorsDetails(errorMessage);
+    const addDatabasePaneFailedMessage = {
+      ...addDatabasePaneMessage,
+      offerThroughput,
+      error: errorMessage,
+      errorStack: getErrorStack(error),
+    };
+    TelemetryProcessor.traceFailure(Action.CreateDatabase, addDatabasePaneFailedMessage, startKey);
+  };
+
+  const _getThroughput = (): number => {
+    return isNaN(throughput) ? 0 : Number(throughput);
+  };
+
+  const _computeOfferThroughput = (): number => {
+    if (!canConfigureThroughput) {
+      return undefined;
+    }
+
+    return _getThroughput();
+  };
+
+  const _isValid = (): boolean => {
+    // TODO add feature flag that disables validation for customers with custom accounts
+    if (isAutoPilotSelected) {
+      const autoPilot = _isAutoPilotSelectedAndWhatTier();
+      if (
+        !autoPilot ||
+        !autoPilot.maxThroughput ||
+        !AutoPilotUtils.isValidAutoPilotThroughput(autoPilot.maxThroughput)
+      ) {
+        setFormErrors(
+          `Please enter a value greater than ${AutoPilotUtils.minAutoPilotThroughput} for autopilot throughput`
+        );
+        return false;
+      }
+    }
+    const throughput = _getThroughput();
+
+    if (throughput > SharedConstants.CollectionCreation.DefaultCollectionRUs100K && !throughputSpendAck) {
+      setFormErrors(`Please acknowledge the estimated daily spend.`);
+      return false;
+    }
+
+    const autoscaleThroughput = maxAutoPilotThroughputSet * 1;
+
+    if (
+      isAutoPilotSelected &&
+      autoscaleThroughput > SharedConstants.CollectionCreation.DefaultCollectionRUs100K &&
+      !throughputSpendAck
+    ) {
+      setFormErrors(`Please acknowledge the estimated monthly spend.`);
+      return false;
+    }
+
+    return true;
+  };
+
+  const handleonChangeDBId = React.useCallback(
+    (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+      setDatabaseId(newValue || "");
+    },
+    []
+  );
+
+  const genericPaneProps: RightPaneFormProps = {
```

Comment:
nit: rename variable name to `genericPaneProps`

Reply:
done